src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -25,26 +27,59 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.time.LocalDate;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.json.JSONException;
+import org.json.JSONObject;
 
 /**
- * Servlet that schedules tasks on tomorrow.
+ * Schedules tasks on tomorrow.
  */
 @WebServlet("/schedule")
 public class ScheduleServlet extends HttpServlet {
+
+  public static final String TASK_ID_LIST_KEY = "taskId";
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    TasksProvider taskProvider = new TasksProvider();
+    List<Task> tasks = taskProvider.getTasks();
+    if(!request.getParameterMap().containsKey(TASK_ID_LIST_KEY)){
+      sendJsonResponse(response, "Select some tasks to schedule.");
+      return;
+    }
+    Set<String> selectedIds = Arrays.stream(request.getParameterValues(TASK_ID_LIST_KEY)).collect(Collectors.toSet());
+    List<String> titlesOfTasksToSchedule = filterSelectedTaskTitles(selectedIds, tasks);
```

Comment:
It would be better to pass a list of tasks rather than just their titles into the Scheduler. Afterall we'll eventually need the description etc of them.

Reply:
I thought that this change is going to happen in a different step, but ok.

Reply:
After the Tasks API integration, I think it makes sense to pass the list of the selected ids and then the scheduler will retrieve each task when it will be scheduled. In that way we can save the retrieval of tasks that will not be scheduled.

Reply:
that sounds reasonable. I leave it to you to decide when you want to refactor this.

Reply:
I think if we make the scheduler retrieve the tasks by Id, it makes testing hard. (It has to access the TaskProvider then.)
Maybe passing an actual list of Task objects to the scheduler is better from this perspective.

Reply:
Yeah, it makes sense.